### PR TITLE
8343411: Test fail on Windows due to "An established connection was aborted by the software in your host machine"

### PR DESCRIPTION
**Pull request by djelinski**
This PR fixes an intermittent test failure on Windows. The test server used in HttpURLConnectionExpect100Test now reads from the socket until the client closes it.

The current implementation closes the server socket after reading the request headers, but without reading the request body. On Windows, if any data is left in the socket receive buffer when the socket is closed, the connection is reset. If the client doesn't read the response before the connection is reset, the response is lost and the test fails.

Verified that:
- the test still passes
- waiting for the client to close the socket does not take excessive amounts of time (the test finishes in half a second on my machine)

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8343411](https://bugs.openjdk.org/browse/JDK-8343411): Test fail on Windows due to "An established connection was aborted by the software in your host machine" (**Bug** - P4)

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21948/head:pull/21948` \
`$ git checkout pull/21948`

Update a local copy of the PR: \
`$ git checkout pull/21948` \
`$ git pull https://git.openjdk.org/jdk.git pull/21948/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21948`

View PR using the GUI difftool: \
`$ git pr show -t 21948`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21948.diff">https://git.openjdk.org/jdk/pull/21948.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21948#issuecomment-2461993036)
</details>

### CODE REVIEW

**Review by dfuch — APPROVED** [review]
Looks reasonable to me. Plus good cleanup with try-with-resources!

File: test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpect100Test.java [hunk 1]
```diff
@@ -230,6 +218,10 @@ private void handle_connection(BufferedReader in, PrintStream out)
             } else {
                 defaultResponse(out);
             }
+            // wait until the client closes the socket
+            while (line != null) {
```
Comment (jaikiran) [inline test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpect100Test.java:222]: Hello Daniel, does the code ever enter this `while` loop? If I'm reading the above few lines correctly, then when the code exits the previous `do while` loop then `line` would be `null` and `line` appears to continue being `null` when it reaches here.

Comment (djelinski) [reply test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpect100Test.java:222]: Hi Jaikiran, the previous write loop is usually exited with a zero-length line. Null is only returned on EOF.

Comment (jaikiran) [reply test/jdk/java/net/HttpURLConnection/HttpURLConnectionExpect100Test.java:222]: Thank you that explains it.

**Review by jaikiran — APPROVED** [review]

**Comment by djelinski** [comment]
Thanks for the reviews!

/integrate

**Pull request closed without merging**